src/Engine/Manager/ScriptManager.cpp

```diff
@@ -58,9 +58,57 @@ void AngelScriptMessageCallback(const asSMessageInfo* message, void* param) {
     Log() << " : " << message->message << "\n";
 }
 
+std::string CallstackToString(asIScriptContext* ctx) {
+
+    std::string callstack = "Callstack:\n";
+    for (asUINT n = 0; n < ctx->GetCallstackSize(); n++) {
+        asIScriptFunction* func;
+        const char* scriptSection;
+        int line, column;
+        func = ctx->GetFunction(n);
+        line = ctx->GetLineNumber(n, &column, &scriptSection);
+        callstack.append(func->GetDeclaration());
+        callstack.append(":");
+        callstack.append(std::to_string(line));
+        callstack.append(",");
+        callstack.append(std::to_string(column));
+        callstack.append("\n");
+    }
+    return callstack;
+}
+std::string VariablesToString(asIScriptContext* ctx, asUINT stackLevel)
+{
+    asIScriptEngine* engine = ctx->GetEngine();
+    // First print the this pointer if this is a class method
+    int typeId = ctx->GetThisTypeId(stackLevel);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'typeId' is assigned a value that is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)